[Leong Jia Hao Daniel] iP

leongdl135 opened the pull request:
### Duke
> "If you're not a good shot today, don't worry. There are other ways to be useful." ~ Alexander (Sasha) Novikov

Duke assists you with remembering your tasks so that you can free your mind. It's
- text-based
- easy to learn
- ~~fast~~ SUPER FAST to use

To begin, all you need to do is: 
1. Download it from [here](https://github.com/leongdl135/ip/releases/tag/A-Jar)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
The best part? IT IS **FREE**

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Search tasks
***
If you Java programmer, you can use it to practice Java too. Here's the 'main' method:
``` java
public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
    }
```

Commit:
Add Gradle support

leongdl135 committed:
Completed Level 1 (Greet, Echo, Exit)

leongdl135 committed:
Completed Level 2 (Add, List)

leongdl135 committed:
Completed Level 3 (Mark as done)

leongdl135 committed:
Added Task Class

leongdl135 committed:
A-Inheritance: Added Deadline, ToDo and Event classes

leongdl135 committed:
Completed Level 4 (ToDos, Events, Deadlines)

leongdl135 committed:
A-TextUiTesting: Added input and expected output txt files

leongdl135 committed:
A-Enums: added command enums

leongdl135 committed:
Completed Level 5 (Handle Errors)

leongdl135 committed:
A-Exceptions: Created exceptions to handle errors

leongdl135 committed:
A-Collections: Use ArrayList<>() to store tasks

leongdl135 committed:
Completed Level-6 (Delete)

leongdl135 committed:
Added new test cases for Level-6 requirements

leongdl135 committed:
Bug Fix: Make enums capitalized

leongdl135 committed:
Level-7: Add Save Feature

leongdl135 committed:
Level-8: Add Date and Time to Deadline

leongdl135 committed:
Merge branch 'branch-Level-7'

leongdl135 committed:
Merge branch-Level-7

leongdl135 committed:
Merge branch 'branch-Level-8'

leongdl135 committed:
Fix merge conflict

leongdl135 committed:
Merge branch 'add-gradle-support'

leongdl135 committed:
Refactor code to add more OOP

leongdl135 committed:
A-Packages: Add packages to code

leongdl135 committed:
Update build.gradle class name

leongdl135 committed:
A-JUnit: Add test cases

leongdl135 committed:
Fix bug add good bye message

leongdl135 committed:
A-JavaDoc: Add javadocs for public classes and methods.

leongdl135 committed:
A-CodingStandard: Fix code style violations.

leongdl135 committed:
Level-9: Add Find command.

leongdl135 committed:
Merge branch 'branch-A-JavaDoc'

leongdl135 committed:
Merge branch 'branch-A-CodingStandard'

leongdl135 committed:
Merge branch 'branch-Level-9'

leongdl135 committed:
Fix test cases

leongdl135 committed:
Refactor method names

leongdl135 committed:
A-Checkstyle: Add checkstyle to gradle

leongdl135 committed:
Fix checkstyle errors

leongdl135 committed:
Fix test file coding standard violations

chloeelim commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,39 @@
+
+    private String keyWord;
+
+    public FindCommand(String keyWord) {
```
Perhaps there should be a header comment for the FindCommand constructor here? [(Reference to style guide: write descriptive header comments for all public methods)](https://se-education.org/guides/conventions/java/index.html#comments)

chloeelim commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,39 @@
+        this.keyWord = keyWord;
+    }
+
+    @Override
```
Should there be a header comment for this overridden `execute` method since its [parent method's Javadoc](https://github.com/nus-cs2103-AY2223S1/ip/pull/229/files#diff-bf1b68644e11c2becb2048f5148320c1fe7095d4f819ab39755280a1c8863c01R16) is not very specific to this overridden method? [(Reference to style guide: write descriptive header comments for all public methods)](https://se-education.org/guides/conventions/java/index.html#comments)

chloeelim commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,39 @@
+        ui.displayMatchingList(temp);
+    }
+
+    @Override
```
Similar to the issue above, perhaps there should there be a header comment for this overridden `isExit` method since its [parent method's Javadoc](https://github.com/nus-cs2103-AY2223S1/ip/pull/229/files#diff-bf1b68644e11c2becb2048f5148320c1fe7095d4f819ab39755280a1c8863c01R26) cannot apply exactly to itself? [(Reference to style guide: write descriptive header comments for all public methods)](https://se-education.org/guides/conventions/java/index.html#comments)

chloeelim commented on src/main/java/duke/exception/InvalidArgumentException.java:
```diff
@@ -0,0 +1,53 @@
+            message += "\nPlease add a /at to declare the time the event is at.";
+            break;
+        case Mark:
+        case Unmark:
```
Should there be a `// Fallthrough` comment for each of these cases without a `break` statement here? [(Reference: The explicit //Fallthrough comment should be included whenever there is a case statement without a break statement.)](https://se-education.org/guides/conventions/java/index.html#conditionals)

chloeelim commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,197 @@
+public class Parser {
+
+    /**
+     * The main parsing command which handles the raw input string from
```
Perhaps you should consider rephrasing this Javadoc header so it starts with a verb + 's' (eg: parses ...)? [(Reference: In method header comments, the first sentence should start in the form Returns..., Sends..., Adds...)](https://se-education.org/guides/conventions/java/index.html#comments)

chloeelim commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,62 @@
+        }
+    }
+
+    public static void main(String[] args) {
```
Maybe consider adding a Javadoc header for the public `main` method? [(Reference: write descriptive header comments for all public classes/methods)](https://se-education.org/guides/conventions/java/index.html#comments)

chloeelim commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,62 @@
+    private Ui ui;
+
+    /**
+     * Constructor for Duke.
```
Should this Javadoc header start with a verb + 's' instead? (eg: "Constructs" or "Returns") [(Reference: In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ...)](https://se-education.org/guides/conventions/java/index.html#comments) (I noticed this issue happens a few times throughout the code)

chloeelim commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,54 @@
+     * @param ui The ui class which handles the user interface.
+     * @param storage The storage class which deals with the file.
+     * @param taskList The tasklist that stores the tasks.
+     * @throws DukeException throws if there is an error.
```
Maybe capitalize the starting of the exception description so it is consistent with the other descriptors?

```suggestion
     * @throws DukeException throws If there is an error.
```

chloeelim commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,54 @@
+    private Task task;
+
+    /**
+     * The constructor for the AddCommand.
```
Should this Javadoc header start with a verb + 's' instead? (eg: "Constructs" or "Returns") [(Reference: In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ...)](https://se-education.org/guides/conventions/java/index.html#comments)

chloeelim commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,57 @@
+     * @param ui The ui class which handles the user interface.
+     * @param storage The storage class which deals with the file.
+     * @param taskList The tasklist that stores the tasks.
+     * @throws DukeException throws if there is an error.
```
Maybe capitalize the starting of the exception description so it is consistent with the other descriptors?

chloeelim commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,42 @@
+     * @param ui The ui class which handles the user interface.
+     * @param storage The storage class which deals with the file.
+     * @param taskList The tasklist that stores the tasks.
+     * @throws DukeException throws if there is an error.
```
Maybe capitalize the starting of the exception description so it is consistent with the other descriptors? (I think this issue happens a few times throughout the code)

leongdl135 committed:
Level 10: Add GUI to duke

chloeelim commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,104 @@
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.ToDo;
+/**
```
Perhaps there should be a line break here? [(Reference)](https://google.github.io/styleguide/javaguide.html#s3-source-file-structure)

chloeelim commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,104 @@
+    /**
+     * The Storage constructor.
+     *
+     * @param filePath
```
Should there be a description for the param `filePath` here?

leongdl135 committed:
Add GUI to Duke

leongdl135 committed:
Merge branch 'branch-Level-10'

chloeelim commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,77 @@
+        String[] details = dateTime.split(" ");
+        DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern("d/MM/yyyy");
+        this.date = LocalDate.parse(details[0], inputFormat);
+        this.time = details[1];
```
I think the use of `this` here is unnecessary since it is not shadowed by a constructor param. [(Reference: Avoid unnecessary use of this with fields)](https://se-education.org/guides/conventions/java/index.html#variables)

```suggestion
        date = LocalDate.parse(details[0], inputFormat);
        time = details[1];
```

chloeelim commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,77 @@
+    @Override
+    public String toDataFormat() {
+        String completed = "0";
+        if (this.getStatusIcon().equals("X")) {
```
Similarly, I think the `this` keyword here might also be unnecessary because it is not shadowed by anything else. [(Reference: Avoid unnecessary use of this with fields)](https://se-education.org/guides/conventions/java/index.html#variables)

```suggestion
        if (getStatusIcon().equals("X")) {
```

leongdl135 committed:
Add Welcome message to Duke

chloeelim commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,64 @@
+    @Override
+    public String toDataFormat() {
+        String completed = "0";
+        if (this.getStatusIcon().equals("X")) {
```
I think the `this` keyword here might also be unnecessary because it is not shadowed by anything else. [(Reference: Avoid unnecessary use of this with fields)](https://se-education.org/guides/conventions/java/index.html#variables) (This issue occurs a few more times throughout the code)

```suggestion
        if (getStatusIcon().equals("X")) {
```

chloeelim commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    /**
+     * Override the toString() method to display the task to the user.
```
Should the Javadocs start with the pluralised form of 'override' instead? [(Reference: In method header comments, the first sentence should start in the form Returns..., Sends..., Adds...)](https://se-education.org/guides/conventions/java/index.html#comments)

```suggestion
     * Overrides the toString() method to display the task to the user.
```

chloeelim commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,70 @@
+    /**
+     * Overrides the String implementation Task
+     *
+     * @return A string that shows the completion status of task
```
Maybe the Javadocs here is missing punctuation? [(Reference: Punctuation behind each parameter description)](https://se-education.org/guides/conventions/java/index.html#comments)

```suggestion
     * Overrides the String implementation Task.
     *
     * @return A string that shows the completion status of task.
   ```

chloeelim commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,86 @@
+     * Another constructor for the taskList when the input file is
+     * not empty.
+     *
+     * @param list
```
Perhaps you're missing the param description for `list` here?

chloeelim commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,117 @@
+import java.util.Scanner;
+
+import duke.task.Task;
+/**
```
Perhaps there should be a line break here (separating the import statements and top-level class)? [(Reference)](https://google.github.io/styleguide/javaguide.html#s3-source-file-structure)

```suggestion

/**
```

chloeelim commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,197 @@
+     * @throws DukeException If the input String has an error.
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] inputArray = input.split(" ", 2);
```
Perhaps consider renaming `inputArray` to something plural? [(Reference: Plural form should be used on names representing a collection of objects)](https://se-education.org/guides/conventions/java/index.html#naming)

chloeelim reviewed (commented):
Just a few minor styling nits to fix, overall I found your code easy to read and it looks like you put in a lot of effort to adhere to the style guide, so really good job! 👍

senwei01 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,86 @@
+package duke.task;
```
I like that there are multiple operations for you to filter the tasks!

senwei01 commented on src/main/java/duke/exception/InvalidArgumentException.java:
```diff
@@ -0,0 +1,53 @@
+package duke.exception;
```
I like how there's many ways that your code handles many different type of exception!

senwei01 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,77 @@
+        super(description);
+        String[] details = dateTime.split(" ");
+        DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern("d/MM/yyyy");
+        this.date = LocalDate.parse(details[0], inputFormat);
```
Perhaps you might want to consider removing the `this` since the parameter you take in your constructor for Deadline has a different name.

senwei01 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,111 @@
+ *
+ * @author Leong Jia Hao Daniel
+ */
+public class Ui {
```
I like that there are multiple ways that your code displays the output based on the user's input

senwei01 reviewed (commented):
Overall, i like your code! I think the readability and OOP are both good!

leongdl135 committed:
Add assertions

There are no assertions in my code.

This needs to be changed as assertions help to define assumptions about
the program state so that the runtime can verify them. Assertions need
to be added to help increase code readability and error detection.

Let's add assetions to the following classes.
- Deadline
- Event
- Task
- ToDo
- DeleteCommand
- MarkCommand
- UnMarkCommand

Using assertions in these classes help to ensure that the commands
receive the right information.

leongdl135 committed:
Refactor code to fit code quality standards

The code does not fit the code quality standards.

Refactoring the code and improving its quality improves code quality and
allows other to read my code more easily. It is also easier to fix bugs
when my code is of a better standard.

Let's update the code to fit the standard.

leongdl135 committed:
Merge pull request #1 from leongdl135/branch-A-Assertions

Add assertions

leongdl135 committed:
Merge branch 'master' into branch-A-CodeQuality

leongdl135 committed:
Merge pull request #2 from leongdl135/branch-A-CodeQuality

Refactor code to fit code quality standards

leongdl135 committed:
A-CI: Set up CI

leongdl135 committed:
Add C-MassOps extension

My commands do not allow users to quickly clear completed tasks/ unmark
tasks efficiently.

This is not user friendly as users who want to clear completed tasks
have to clear them one at a time and it takes quite a lot of effort to
do so.

Let's add some new commands.
- Mass Delete Command
- Mass UnMark Command

These are the first few mass commands I added as these are the most
important features for now.

leongdl135 committed:
Merge pull request #3 from leongdl135/branch-C-MassOps

Add C-MassOps extension

leongdl135 committed:
Fix checkstyle issues

leongdl135 committed:
Add UserGuide

leongdl135 committed:
Add JavaDocs to method headers

leongdl135 committed:
Fix massunmark bug

leongdl135 committed:
Update mainClassName in build.gradle

leongdl135 committed:
Fix tasks bugs

leongdl135 committed:
Fix window rezise issue

leongdl135 committed:
Refactor error message

leongdl135 committed:
Update test cases

leongdl135 committed:
Fix exit bug
